### PR TITLE
feat: worker listening-tentacle create

### PR DESCRIPTION
**Pull request by benPearce1**
feat: worker ssh create

[sc-29479]

ssh help:
image

ssh create interactive:
image

ssh create automation:
image

listening tentacle create help
image

listening-tentacle create interactive
image

listening tentacle create automation
image

### CODE REVIEW

File: pkg/cmd/worker/listening-tentacle/create/create.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		return err
+	}
+	worker.MachinePolicyID = machinePolicy.GetID()
+
+	createdWorker, err := opts.Client.Workers.Add(worker)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(opts.Out, "Successfully created listening tentcle worker '%s'.\n", worker.Name)
```
Comment (jbristowe) [inline pkg/cmd/worker/listening-tentacle/create/create.go:133]:
```suggestion
	fmt.Fprintf(opts.Out, "Successfully created Listening Tentacle worker '%s'.\n", worker.Name)
```

File: pkg/cmd/worker/listening-tentacle/create/create.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+			opts := NewCreateOptions(createFlags, cmd.NewDependencies(f, c))
+
+			return createRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&createFlags.Name.Value, createFlags.Name.Name, "n", "", "A short, memorable, unique name for this Listening Tentacle worker.")
+	flags.StringVar(&createFlags.Thumbprint.Value, createFlags.Thumbprint.Name, "", "The X509 certificate thumbprint that securely identifies the Tentacle.")
+	flags.StringVar(&createFlags.URL.Value, createFlags.URL.Name, "", "The network address at which the Tentacle can be reached.")
```
Comment (jbristowe) [inline pkg/cmd/worker/listening-tentacle/create/create.go:85]:
```suggestion
	flags.StringVar(&createFlags.URL.Value, createFlags.URL.Name, "", "The network address at which the Listening Tentacle can be reached.")
```

File: pkg/cmd/worker/listening-tentacle/create/create.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+	err = machinescommon.PromptForMachinePolicy(opts.CreateTargetMachinePolicyOptions, opts.CreateTargetMachinePolicyFlags)
+	if err != nil {
+		return err
+	}
+
+	if opts.Thumbprint.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Thumbprint",
+			Help:    "The X509 certificate thumbprint that securely identifies the Tentacle.",
```
Comment (jbristowe) [inline pkg/cmd/worker/listening-tentacle/create/create.go:163]:
```suggestion
			Help:    "The X509 certificate thumbprint that securely identifies the Listening Tentacle.",
```

File: pkg/cmd/worker/listening-tentacle/create/create.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+			survey.MaxLength(40),
+		))); err != nil {
+			return err
+		}
+	}
+
+	if opts.URL.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "URL",
+			Help:    "The network address at which the Tentacle can be reached.",
```
Comment (jbristowe) [inline pkg/cmd/worker/listening-tentacle/create/create.go:175]:
```suggestion
			Help:    "The network address at which the Listening Tentacle can be reached.",
```

File: pkg/cmd/worker/listening-tentacle/listening-tentacle.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	cmdCreate "github.com/OctopusDeploy/cli/pkg/cmd/worker/listening-tentacle/create"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdListeningTentacle(f factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "listening-tentacle <command>",
+		Short:   "Manage listening tentacle workers",
```
Comment (jbristowe) [inline pkg/cmd/worker/listening-tentacle/listening-tentacle.go:15]:
```suggestion
		Short:   "Manage Listening Tentacle workers",
```

File: pkg/cmd/worker/listening-tentacle/listening-tentacle.go [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+		Long:    "Work with Octopus Deploy listening tentacle workers.",
```
Comment (jbristowe) [inline pkg/cmd/worker/listening-tentacle/listening-tentacle.go:16]:
```suggestion
		Long:    "Work with Listening Tentacle workers in Octopus Deploy.",
```

File: pkg/cmd/worker/shared/workerpool.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func NewWorkerPoolFlags() *WorkerPoolFlags {
+	return &WorkerPoolFlags{
+		WorkerPools: flag.New[[]string](FlagWorkerPool, false),
+	}
+}
+
+func RegisterCreateWorkerWorkerPoolFlags(cmd *cobra.Command, flags *WorkerPoolFlags) {
+	cmd.Flags().StringSliceVar(&flags.WorkerPools.Value, flags.WorkerPools.Name, []string{}, "The worker pools which the worker will be a member of")
```
Comment (jbristowe) [inline pkg/cmd/worker/shared/workerpool.go:33]:
```suggestion
	cmd.Flags().StringSliceVar(&flags.WorkerPools.Value, flags.WorkerPools.Name, []string{}, "The pools of which the worker will be a member.")
```

File: pkg/cmd/worker/shared/workerpool.go [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+	return ids, nil
+}
+
+func PromptForWorkerPools(opts *WorkerPoolOptions, flags *WorkerPoolFlags) error {
+	if util.Empty(flags.WorkerPools.Value) {
+		allWorkerPools, err := opts.GetAllWorkerPoolsCallback()
+		if err != nil {
+			return err
+		}
+		selectedPools, err := question.MultiSelectMap(opts.Ask, "Select the worker pools to assign the worker to", allWorkerPools, func(pool *workerpools.WorkerPoolListResult) string { return pool.Name }, true)
```
Comment (jbristowe) [inline pkg/cmd/worker/shared/workerpool.go:85]:
```suggestion
		selectedPools, err := question.MultiSelectMap(opts.Ask, "Select the worker pools to assign to the worker.", allWorkerPools, func(pool *workerpools.WorkerPoolListResult) string { return pool.Name }, true)
```

File: pkg/cmd/worker/shared/workerpool_test.go [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+	opts := shared.NewWorkerPoolOptions(&cmd.Dependencies{Ask: asker})
+	err := shared.PromptForWorkerPools(opts, flags)
+	checkRemainingPrompts()
+	assert.NoError(t, err)
+}
+
+func TestPromptForWorkerPool_NoFlagsSupplied(t *testing.T) {
+	pa := []*testutil.PA{
+		testutil.NewMultiSelectPrompt("Select the worker pools to assign the worker to", "", []string{"Groundskeeper", "Swim instructor"}, []string{"Groundskeeper", "Swim instructor"}),
```
Comment (jbristowe) [inline pkg/cmd/worker/shared/workerpool_test.go:27]:
```suggestion
		testutil.NewMultiSelectPrompt("Select the worker pools to assign to the worker", "", []string{"Groundskeeper", "Swim instructor"}, []string{"Groundskeeper", "Swim instructor"}),
```

Comment (jbristowe) [inline pkg/cmd/worker/shared/workerpool_test.go:27]:
```suggestion
		testutil.NewMultiSelectPrompt("Select the worker pools to assign to the worker", "", []string{"Groundskeeper", "Swim instructor"}, []string{"Groundskeeper", "Swim instructor"}),
```

File: pkg/cmd/worker/ssh/ssh.go [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdSsh(f factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "ssh <command>",
+		Short:   "Manage SSH workers",
+		Long:    "Work with Octopus Deploy SSH workers.",
```
Comment (jbristowe) [inline pkg/cmd/worker/ssh/ssh.go:16]:
```suggestion
		Long:    "Work with SSH workers in Octopus Deploy.",
```

File: pkg/cmd/worker/worker.go [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/OctopusDeploy/cli/pkg/constants/annotations"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdWorker(f factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "worker <command>",
+		Short: "Manage workers",
+		Long:  `Work with Octopus Deploy workers.`,
```
Comment (jbristowe) [inline pkg/cmd/worker/worker.go:18]:
```suggestion
		Long:  `Work with workers in Octopus Deploy.`,
```

File: pkg/cmd/worker/worker.go [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+		Example: fmt.Sprintf(heredoc.Doc(`
+			$ %s worker list
+			$ %s tenant ls
```
Comment (jbristowe) [inline pkg/cmd/worker/worker.go:21]:
```suggestion
			$ %s worker ls
```

**Review by jbristowe — CHANGES_REQUESTED** [review]
Minor edits, otherwise LGTM.

File: pkg/cmd/worker/worker.go (hunk 12 above)
Comment (benPearce1) [reply pkg/cmd/worker/worker.go:21]: 🤦

**Review by jbristowe — APPROVED** [review]

**Comment by benPearce1** [comment]
BEGIN_COMMIT_OVERRIDE
feat: worker listening-tentacle create
feat: worker ssh create
END_COMMIT_OVERRIDE

**Pull request merged**